client/client.go

```diff
@@ -511,3 +549,44 @@ func (client *Client) DebugGet(action string, result interface{}, params map[str
 	_, err := client.doSync("GET", "/v1/debug", urlParams, nil, nil, &result)
 	return err
 }
+
+type defaultRequester struct {
+	baseURL   url.URL
+	doer      doer
+	userAgent string
+	transport *http.Transport
+	client    *Client
+}
+
+func newDefaultRequester(client *Client, opts *Config) (*defaultRequester, error) {
+	if opts == nil {
+		opts = &Config{}
+	}
+
+	var requester *defaultRequester
+
+	if opts.BaseURL == "" {
+		// By default talk over a unix socket.
+		transport := &http.Transport{Dial: unixDialer(opts.Socket), DisableKeepAlives: opts.DisableKeepAlive}
+		baseURL := url.URL{Scheme: "http", Host: "localhost"}
+		requester = &defaultRequester{baseURL: baseURL, transport: transport}
+	} else {
+		// Otherwise talk regular HTTP-over-TCP.
+		baseURL, err := url.Parse(opts.BaseURL)
+		if err != nil {
+			return nil, fmt.Errorf("cannot parse base URL: %v", err)
```

Comment:
nit:
```suggestion
			return nil, fmt.Errorf("cannot parse base URL: %w", err)
```

Reply:
Looks like this file consistently prefers wrapping of errors, so I've changed this as per Angel's suggestion to align.